### PR TITLE
Add initial native Windows support.

### PR DESCRIPTION
**Pull request by Anakonda**
Rebased and squashed all the commits from  #947 as requested @ThomasWaldmann

### CODE REVIEW

File: borg/helpers.py [hunk 1]
```diff
@@ -2,17 +2,23 @@
 from binascii import hexlify
 from collections import namedtuple, deque
 from functools import wraps, partial
-import grp
+import sys
+if sys.platform != 'win32':
+    import grp
+    import pwd
+else:
+    import encodings.idna
```
Comment (ThomasWaldmann) [inline borg/helpers.py]: do you use this anywhere?

File: docs/development.rst [hunk 2]
```diff
@@ -149,6 +166,9 @@ If you encounter issues, see also our `Vagrantfile` for details.
           work on same OS, same architecture (x86 32bit, amd64 64bit)
           without external dependencies.
 
+On Windows use `python buildwin32.py` To build standalone executable in `win32exe` directory
```
Comment (ThomasWaldmann) [inline docs/development.rst]: lowercase "to"

File: docs/installation.rst [hunk 3]
```diff
@@ -102,6 +102,12 @@ You can change the temporary directory by setting the ``TEMP`` environment varia
 If a new version is released, you will have to manually download it and replace
 the old version using the same steps as shown above.
 
+Windows zip
++++++++++++
+Tested on Windows10. (Should work on Vista and up)
+
+To install on Windows just extract the zip anywhere and add the bin directory to your ``PATH`` environment variable
```
Comment (ThomasWaldmann) [inline docs/installation.rst]: add a dot at the end.

File: docs/installation.rst [hunk 4]
```diff
@@ -200,6 +206,14 @@ and commands to make fuse work for using the mount command.
      sysctl vfs.usermount=1
     
 
+Windows
++++++++
+
+See development_ on how to build on windows.
+run `python3 buildwin32.py` to create standalone windows executable in `win32exe`.
+You can rename or move that folder. Add the bin folder to your ``PATH`` and you can run ``borg``
```
Comment (ThomasWaldmann) [inline docs/installation.rst]: dot

File: setup.py [hunk 5]
```diff
@@ -106,7 +107,22 @@ def detect_lz4(prefixes):
 include_dirs = []
 library_dirs = []
 
-possible_openssl_prefixes = ['/usr', '/usr/local', '/usr/local/opt/openssl', '/usr/local/ssl', '/usr/local/openssl', '/usr/local/borg', '/opt/local']
+windowsIncludeDirs = []
+if sys.platform == 'win32':
+    gccpath = ""
+    for p in os.environ["PATH"].split(";"):
+        if os.path.exists(p + "/gcc.exe"):
```
Comment (ThomasWaldmann) [inline setup.py]: use os.path.join

**Comment by ThomasWaldmann** [comment]
somehow i found some more minor stuff. either i didn't see it in last review or you didn't see my feedback.

**Comment by Anakonda** [comment]
The encoding was for the standalone (which for some reason no longer needs it.)

I did force push to keep it only one commit.

**Pull request merged**